packages/connections/src/stores/connections-store.ts

```diff
@@ -141,6 +146,49 @@ async function loadConnections(
   }
 }
 
+function trackConnectionAttemptEvent({ favorite, lastUsed }: ConnectionInfo): void {
+  const trackEvent = {
+    is_favorite: Boolean(favorite),
+    is_recent: Boolean(lastUsed && !favorite),
+    is_new: !lastUsed,
+  };
+  track('Connection Attempt', trackEvent);
+}
+
+async function trackNewConnectionEvent(dataService: DataService, connectionString: string): Promise<void> {
+  const {
+    dataLake,
+    genuineMongoDB,
+    host,
+    build,
+  } = await dataService.instance();
```

Comment:
this one can be intense .. at least it was till some time ago, @gribnoysup ideas here?

Reply:
It's not anymore in current main (databases and collections are not fetched in the instance method anymore), but it still feels a bit weird to me that we fetch this info twice at the same time in two different places. If we need instance info to report this metric, would it make sense to do it in the place where we already fetch instance info, or are we explicitly trying to keep this metric code scoped to the connections package for a reason (e.g., because it will be used in vscode or something)?

Reply:
are those fetched once per connection? if that's the case probably is easier to emit this entire event there, the `ConnectionOption` object, along with the rest of the properties that we need for telemetry should be available anywhere we call `dataService.instance()` already

Reply:
They are fetched every time the global refresh is activated (either with a 🔄 button or with `cmd+r`)

Reply:
any chance is easy to detect the first time?

Reply:
@gribnoysup I like this idea of emitting this tracking after `refreshInstance`, but then it will be hidden deep inside. reason for having it at the current place - its visible and easier to work with (i had no vscode in my mind).